modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/property/PropertiesHelper.java

```diff
@@ -42,6 +43,23 @@ public static Builder newBuilder() {
         return new BuilderImpl();
     }
 
+    /**
+     * Creates new builder and populates it with properties from the given Map. Keys of the map are names of properties, values are values.
+     *
+     * @param props A holder to populate new builder with.
+     * @return A builder containing properties from given holder.
+     */
+    public static Builder newBuilder(Map<String, Object> props) {
+        Builder builder = newBuilder();
+
+        for (Entry<String, Object> p : props.entrySet()) {
+            Property property = new Property(p.getKey(), p.getValue().getClass());
+            builder.set(property, p.getValue());
+        }
```

Comment:
I don't like the idea to return partially constructed object.
I'd prefer to have one or both methods:
`static Properties fromMap(Map<> props)`
or in Builder class `Builder addFromMap(Map<> props)`